File: jvm/src/main/java/ch/neukom/advent2022/day5/Part1.java
package ch.neukom.advent2022.day5;

import java.io.IOException;

import ch.neukom.advent2022.day5.Storage.StorageType;
import ch.neukom.advent2022.util.InputResourceReader;

public class Part1 {
    public static void main(String[] args) throws IOException {
        try (InputResourceReader reader = new InputResourceReader(Part1.class)) {
            run(reader);
        }
    }

    private static void run(InputResourceReader reader) {
        Util.solvePuzzle(reader, StorageType.CrateMover9000);
    }
}

File: jvm/src/main/kotlin/ch/neukom/advent2022/day10/Part1.kt
package ch.neukom.advent2022.day10

import ch.neukom.advent2022.util.InputResourceReader

object Part1

fun main() {
    InputResourceReader(Part1.javaClass).use(::run)
}

private fun run(reader: InputResourceReader) {
    val finalState = Sequence { reader.readInput().iterator() }
        .map { Instruction.parse(it) }
        .fold(Cpu()) { cpu, instruction -> instruction.execute(cpu) }
    val sumOfSignalStrengths = finalState.getRegisterHistory()
        .mapIndexed { index, register -> (index + 1) * register }
        .filterIndexed { index, _ -> (index + 1 - 20) % 40 == 0 }
        .sum()
    println("The sum of the signal strengths is $sumOfSignalStrengths")
}
